test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1

```diff
@@ -159,8 +159,90 @@ Describe "Get-Content" -Tags "CI" {
+        } | ShouldBeErrorId "TailAndHeadCannotCoexist,Microsoft.PowerShell.Commands.GetContentCommand"
+    }
+
+    It "Should throw TailNotSupported when -Tail used with an unsupported provider" {
+        Push-Location env:
+        {
+        Get-Content PATH -Tail 1 -ErrorAction Stop
+        } | ShouldBeErrorId "TailNotSupported,Microsoft.PowerShell.Commands.GetContentCommand"
+        Pop-Location
+    }
+
+    It "Should throw InvalidOperation when -Tail and -Raw are used" {
+        {
+        Get-Content -Path $testPath -Tail 1 -ErrorAction Stop -Raw
+        } | ShouldBeErrorId "InvalidOperation,Microsoft.PowerShell.Commands.GetContentCommand"
+    }
+    Context "Check Get-Content containing multi-byte chars" {
+        BeforeAll {
+            $firstLine = "Hello,World"
+            $secondLine = "Hello2,World2"
+            $thirdLine = "Hello3,World3"
+            $fourthLine = "Hello4,World4"
+            $fileContent = $firstLine,$secondLine,$thirdLine,$fourthLine
+        }
+        BeforeEach{
+            Set-content -Path $testPath $fileContent
+        }
+        It "Should return all lines when -Tail value is more than number of lines in the file"{
+            $result = Get-Content -Path $testPath -ReadCount -1 -Tail 5 -Encoding UTF7
+            $result.Length | Should Be 4
+            $expected = $fileContent
+            Compare-Object -ReferenceObject $expected -DifferenceObject $result | Should BeNullOrEmpty
+        }
+        It "Should return last three lines at one time for -ReadCount 0 and -Tail 3"{
+            $result = Get-Content -Path $testPath -ReadCount 0 -Tail 3 -Encoding UTF7
+            $result.Length | Should Be 3
+            $expected = $secondLine,$thirdLine,$fourthLine
+            Compare-Object -ReferenceObject $expected -DifferenceObject $result | Should BeNullOrEmpty
+        }
+        It "Should return last three lines reading one at a time for -ReadCount 1 and -Tail 3"{
+            $result = Get-Content -Path $testPath -ReadCount 1 -Tail 3 -Encoding UTF7
+            $result.Length | Should Be 3
+            $expected = $secondLine,$thirdLine,$fourthLine
+            Compare-Object -ReferenceObject $expected -DifferenceObject $result | Should BeNullOrEmpty
+        }
+        It "Should return last three lines at one time for -ReadCount 99999 and -Tail 3"{
+            $result = Get-Content -Path $testPath -ReadCount 99999 -Tail 3 -Encoding UTF7
+            $result.Length | Should Be 3
+            $expected = $secondLine,$thirdLine,$fourthLine
+            Compare-Object -ReferenceObject $expected -DifferenceObject $result | Should BeNullOrEmpty
+        }
+        It "Should return last three lines two lines at a time for -ReadCount 2 and -Tail 3"{
+            $result = Get-Content -Path $testPath -ReadCount 2 -Tail 3 -Encoding UTF7
+            $result.Length | Should Be 2
+            $expected = $secondLine,$thirdLine
+            $expected = $expected,$fourthLine
```

Comment:
Can we use one line `$expected = $secondLine,$thirdLine,$fourthLine `?

Reply:
The result will be a jagged Array, the first element is an array containing two strings (second and third lines) and the second element is a plain string (fourth line) and $expected has to be the same to compare.